### PR TITLE
[kotlin-spring] Adding DocumentationProvider and SwaggerUI

### PR DESCRIPTION
**Pull request by sjoblomj**
This is my Kotlin Spring implementation of @cachescrubber's excellent work in #11258 and #11531.

A user can now select a Documentation Provider (None, Source, Springfox, Springdoc), an Annotation Library (None, Swagger1, Swagger2) and whether to use Swagger UI.

Please note that this removes the `swaggerAnnotations` option that previously existed! Considering this, it is probably a good idea to include this PR in version 6.0.0 with breaking changes, if the PR is deemed to be useful and of sufficient quality.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber @wing328 @jimschubert  @dr4ke616 @karismann @Zomzog @andrewemery @4brunu @yutaka0m

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/kotlin-spring/openapiDocumentationConfig.mustache [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            .licenseUrl("{{licenseUrl}}")
+            .termsOfServiceUrl("{{infoUrl}}")
+            .version("{{appVersion}}")
+            .contact(Contact("", "", "{{infoEmail}}"))
+            .build()
+    }
+
+    @Bean
+{{=<% %>=}}
+    fun customImplementation(servletContext: ServletContext, @Value("${openapi.<%title%>.base-path:<%>defaultBasePath%>}") basePath: String): Docket {
```
Comment (sjoblomj) [inline modules/openapi-generator/src/main/resources/kotlin-spring/openapiDocumentationConfig.mustache:40]: I'm a little unsure about `"${openapi.<%title%>.base-path:<%>defaultBasePath%>}"`.

IntelliJ tells me that it is an unresolved reference, and `defaultBasePath` seems to be empty.

**Comment by sjoblomj** [comment]
[The build is failing](https://github.com/OpenAPITools/openapi-generator/runs/6118624398?check_suite_focus=true) and I don't understand why. It seems upset about new samples that I added:
```
	modified:   samples/server/petstore/kotlin-springboot-source-swagger1/.openapi-generator/FILES
	modified:   samples/server/petstore/kotlin-springboot-source-swagger2/.openapi-generator/FILES
	modified:   samples/server/petstore/kotlin-springboot-springfox/.openapi-generator/FILES
```
Can anyone assist me in the right direction?

**Comment by cachescrubber** [comment]
@sjoblomj Thanks for your effort - I'll try provide more detailed feedback in the next days.

**Comment by wing328** [comment]
LGTM. Thanks for the PR 👍

**Comment by cachescrubber** [comment]
No problem. Just skimmed over the changes - LGTM too. @sjoblomj - if you like please upgrade the Spring Boot version used in the generated poms / build.gradle files? They are still 2.2.0M3.

And the following REQ does apply to the spring kotlind generator too: https://github.com/OpenAPITools/openapi-generator/issues/12220.

**Comment by sjoblomj** [comment]
> No problem. Just skimmed over the changes - LGTM too. @sjoblomj - if you like please upgrade the Spring Boot version used in the generated poms / build.gradle files? They are still 2.2.0M3.
> 
> And the following REQ does apply to the spring kotlind generator too: #12220.

I've made a PR to update dependencies here: #12281

I'll take a look at #12220 and try to make a separate PR for it when I have time :+1:

**Comment by sjoblomjsqueed** [comment]
The problem described in #12220 is addressed in #12377.

**Comment by sdewinter-maqqie** [comment]
Since you have used hardcoded the content type as schema in for example [`apiInterface.mustache`](https://github.com/OpenAPITools/openapi-generator/pull/12184/files#diff-5b0fc09952d467cbf14ab45395c0c1008b745979a39ed70010c52fa3a74e5258R78). It is not possible to generate content schemas for arrays. 

This also is/was an issue in the `java-spring` and seems like #13332 is an bug report for this.

**Pull request merged**